File: JavaOpenDataAccess/src/OpenDataAccess/Service/IEntityRequest.java
package OpenDataAccess.Service;

import OpenDataAccess.Data.IQuery;
import OpenDataAccess.Entity.EntityType;
import OpenDataAccess.Entity.IEntityAccess;
import OpenDataAccess.Entity.IEntityData;
import OpenDataAccess.Entity.IValidate;
import OpenDataAccess.LambdaInterface.IFunction1;

import java.util.Map;

public interface IEntityRequest extends IEntityAccess,IValidate {
    QueryRequest GetQueryRequest();

    void SetQueryRequest(QueryRequest value);

    Request GetRequest();

    void SetRequest(Request value);

    IFunction1<IQuery, IQuery> GetExpandSelectQuery();

    void SetExpandSelectQuery(IFunction1<IQuery, IQuery> value);

    Map<String, Object> GetExceptionDict(String message);

    Map<String, Object> GetMessageDict(String message);

    Map<String, Object> GetBoolDict(boolean blSucceed);

    String CompareVersion(IEntityData newEntityData, IEntityData oldEntityData);

    boolean JudgeIsEdit(EntityType entityType, IEntityData newEntityData, IEntityData oldEntityData) throws  Exception;

    Object Select();

    Object Insert();

    Object Update();

    Object Delete();
}

File: StoryTestCases/OpenDataAccess/OpenDataAccess.Service/IEntityRequest.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Collections.Specialized;
using OpenDataAccess.Data;
using OpenDataAccess.Entity;
using System.Data;
using System.IO;

namespace OpenDataAccess.Service
{
    public interface IEntityRequest : IEntityAccess, IValidate
    {
        QueryRequest _QueryRequest { get; set; }
        Request _Request { get; set; }
        Func<IQuery, IQuery> ExpandSelectQuery { get; set; }
        Dictionary<string, object> GetExceptionDict(string message);
        Dictionary<string, object> GetMessageDict(string message);
        Dictionary<string, object> GetBoolDict(bool blSucceed);
        string CompareVersion(IEntityData newEntityData, IEntityData oldEntityData);
        bool JudgeIsEdit(EntityType entityType, IEntityData newEntityData, IEntityData oldEntityData);
        object Select();
        object Insert();
        object Update();
        object Delete();
    }
}
